src/app/participate/deposit-data-validator.ts

```diff
@@ -11,7 +13,8 @@ import { AbstractControl, NG_VALIDATORS, Validator, ValidatorFn, ValidationError
 })
 export class DepositDataValidatorDirective implements Validator {
   validate(control: AbstractControl): ValidationErrors|null {
-    if (/^0x[0-9a-f]+$/.test((control.value || '').trim())) {
+    console.log(control.value.length);
+    if (/^0x[0-9a-f]+$/.test((control.value || '').trim()) && control.value.trim().length === DEPOSIT_DATA_LENGTH) {
```

Comment:
If value is null or undefined then trim will throw. 

You should do (control.value || ‘’).trim()